jax/experimental/mosaic/gpu/profiler.py

```diff
@@ -109,6 +98,90 @@ def run(*args, **kwargs):
   return outs, float(elapsed)
 
 
+def _measure_cupti(f, aggregate):
+  def wrapper(*args, **kwargs):
+    mosaic_gpu_lib._mosaic_gpu_ext._cupti_init()
+    try:
+      results = jax.block_until_ready(jax.jit(f)(*args, **kwargs))
+    finally:
+      timings = mosaic_gpu_lib._mosaic_gpu_ext._cupti_get_timings()
+    if not timings:
+      return results, None
+    elif aggregate:
+      return results, sum(item[1] for item in timings)
+    else:
+      return results, timings
+  return wrapper
+
+
+def measure(f: Callable, *, mode: str = "cupti", aggregate: bool = True):
```

Comment:
FYI you can be a bit more precise with the types here:

```
def measure(f: Callable[P, T], ...) -> Callable[P, tuple[T, float]]: ...
```

Reply:
That return type would be incorrect when `aggregate=False`. Wouldn't hurt to add `-> Callable` return type though, right?

Reply:
Added `-> Callable`.

Reply:
Well, if you really want it, you can define overloads for `measure` with literal types for `aggregate`, but I have mixed feelings about having different return types tbh as mentioned in the other thread.

Reply:
> I have mixed feelings about having different return types

Why though? The default value of `aggregate` is `True`, which means for both modes the return types are the same. The user needs to consciously type out `aggregate=False` to actually get that array of tuples instead of the default aggregate value.